code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -132,14 +133,22 @@
 	qdel(reagent_puff)
 
 /obj/item/reagent_containers/spray/attack_self(mob/user)
+	..()
+	toggle_stream_mode(user)
+
+/obj/item/reagent_containers/spray/attack_self_secondary(mob/user)
+	..()
```

Comment:
```suggestion
	. = ..()
	toggle_stream_mode(user)

/obj/item/reagent_containers/spray/attack_self_secondary(mob/user)
	. = ..()
```
* Preserve parent call return by default, unless if you have a good reason not to.

Reply:
I don't even know what this does lol

I get that . is what we return with because I see and add stuff like `. += "<span class='notice'>Ctrl+click to toggle power.</span>"` under `examine()` where it's adding to a list to eventually format and send to the player

And I think that if you went `return ..()` here you'd stop the rest of the code executing

But what is the difference between `. = ..()` and `..()`? Does one call the parent proc immediately and the other wait until the rest of the code is executed or something?

Reply:
It preserves the parent call return.
`..()` will call the parent proc, which will return some kind of value, `null` if nothing else.
That value will be lost if not preserved somewhere, and that's that the dot variable is doing: passing it forward.

We try to orient our code so that parent returns trickle down the line, and redefinitions that call parent are mindful of it.